desc/equilibrium/coords.py

```diff
@@ -118,29 +133,20 @@ def map_coordinates(  # noqa: C901
                 **kwargs,
             )
 
-    basis_derivs = tuple(f"{X}_{d}" for X in inbasis for d in ("r", "t", "z"))
-    for key in basis_derivs:
-        errorif(
-            key not in data_index["desc.equilibrium.equilibrium.Equilibrium"],
-            NotImplementedError,
-            f"don't have recipe to compute partial derivative {key}",
-        )
+    # do surface average to get iota once
+    if "iota" in profiles and profiles["iota"] is None:
+        profiles["iota"] = eq.get_profile(["iota", "iota_r"], params=params)
```

Comment:
Is iota_r being used anywhere or is it just there so you can use Hermite?

Reply:
`iota` and `iota_r` get computed on `QuadratureGrid` then we use those values to construct Hermite spline for $\iota(\rho)$. $\iota(\rho)$ is then evaluated at whatever sequence of $\rho$ values the Newton iteration proceeds with. For $\rho, \alpha, \zeta$ coordinate mappings, the $\rho$ values don't need to change throughout Newton iteration since we can just do 1D root finding, which this PR now ensures is done.